lib/rubyplot/backend/magick_wrapper.rb

```diff
@@ -19,10 +19,354 @@ class MagickWrapper < Base
+          draw.fill Rubyplot::Color::COLOR_INDEX[fill_color]
+          draw.polyline(
+            x + size/2, y + size/2, x + size/2, y - size/2,
+            x + size/2, y - size/2, x - size/2, y,
+            x - size/2, y, x + size/2, y + size/2
+          )
+        },
+        solid_tri_left: ->(draw, x, y, fill_color, border_color, size) {
+          # height and base are equal to size
+          # x,y is center of base and height
+          draw.stroke Rubyplot::Color::COLOR_INDEX[fill_color]
+          draw.fill Rubyplot::Color::COLOR_INDEX[fill_color]
+          draw.polyline(
+            x - size/2, y - size/2, x - size/2, y + size/2,
+            x - size/2, y + size/2, x + size/2, y,
+            x + size/2, y, x - size/2, y - size/2
+          )
+        },
+        hollow_plus: ->(draw, x, y, fill_color, border_color, size) {
+          # height and width are equal to size
+          # x,y is center of width and height
+          draw.stroke Rubyplot::Color::COLOR_INDEX[fill_color]
+          draw.fill_opacity 0
+          draw.polyline(
+            x + size/4, y- size/2, x + size/4, y - size/4,
+            x + size/4, y - size/4, x + size/2, y - size/4,
+            x + size/2, y - size/4, x + size/2, y + size/4,
+            x + size/2, y + size/4, x + size/4, y + size/4,
+            x + size/4, y + size/4, x + size/4, y + size/2,
+            x + size/4, y + size/2, x - size/4, y + size/2,
+            x - size/4, y + size/2, x - size/4, y + size/4,
+            x - size/4, y + size/4, x - size/2, y + size/4,
+            x - size/2, y + size/4, x - size/2, y - size/4,
+            x - size/2, y - size/4, x - size/4, y - size/4,
+            x - size/4, y - size/4, x - size/4, y - size/2,
+            x - size/4, y - size/2, x + size/4, y - size/2
+          )
+          draw.fill_opacity 1
+        },
+        solid_plus: ->(draw, x, y, fill_color, border_color, size) {
+          # height and width are equal to size
+          # x,y is center of width and height
+          draw.stroke Rubyplot::Color::COLOR_INDEX[border_color]
+          draw.fill Rubyplot::Color::COLOR_INDEX[fill_color]
+          draw.polyline(
+            x + size/4, y - size/2, x + size/4, y - size/4,
+            x + size/4, y - size/4, x + size/2, y - size/4,
+            x + size/2, y - size/4, x + size/2, y + size/4,
+            x + size/2, y + size/4, x + size/4, y + size/4,
+            x + size/4, y + size/4, x + size/4, y + size/2,
+            x + size/4, y + size/2, x - size/4, y + size/2,
+            x - size/4, y + size/2, x - size/4, y + size/4,
+            x - size/4, y + size/4, x - size/2, y + size/4,
+            x - size/2, y + size/4, x - size/2, y - size/4,
+            x - size/2, y - size/4, x - size/4, y - size/4,
+            x - size/4, y - size/4, x - size/4, y - size/2,
+            x - size/4, y - size/2, x + size/4, y - size/2
+          )
+        },
+        pentagon: ->(draw, x, y, fill_color, border_color, size) {
```

Comment:
Put a `raise NotImplementedError, "marker #{marker_name} has not yet been implemented"` inside each of the lambdas that aren't yet implemented.

Reply:
[[Commit]](https://github.com/SciRuby/rubyplot/pull/45/commits/e2d6541db9a0c94be265f81946909721623e1e6d)
Done but could not add the marker name in the exception being raised because I found no way to get the key name in the value lambda.  
Link to stack overtflow discussion - [Link](https://stackoverflow.com/questions/56639642/how-to-get-the-key-name-inside-the-value-for-printing-inside-a-ruby-hash)  
  
Please suggest if there is anyway to do this otherwise it's fine.

Reply:
You can just write it literally inside the string for now.